dashboards/pubsub/pubsub_topic_detail.sp

```diff
@@ -0,0 +1,503 @@
+        'Name', i.name,
+        'Role ID', i.role_id,
+        'Location', i.location,
+        'Project', i.project,
+        'Stage', i.stage,
+        'Description', i.description
+      ) as properties
+    from
+      iam_role as t join gcp_iam_role i on t.role = i.name
+    where 
+      t.name = $1;
+  EOQ
+
+  param "name" {}
+}
+
+edge "gcp_pubsub_topic_to_iam_role_edge" {
+  title = "assigned to"
+
+  sql = <<-EOQ
+  with iam_role as (
+    select 
+      t.name,
+      roles->>'role' as role 
+    from 
+      gcp_pubsub_topic t, 
+      jsonb_array_elements(t.iam_policy->'bindings') as roles 
+  )
+    select
+      t.name as from_id,
+      i.role_id as to_id,
+      jsonb_build_object(
+        'Name', i.name,
+        'Role ID', i.role_id,
+        'Location', i.location,
+        'Project', i.project,
+        'Stage', i.stage,
+        'Description', i.description
+      ) as properties
+    from
+      iam_role as t join gcp_iam_role i on t.role = i.name
+    where 
+      t.name = $1;
+  EOQ
+
+  param "name" {}
+}
+
+node "gcp_pubsub_topic_to_pubsub_subscription_node" {
+  category = category.gcp_pubsub_subscription
+
+  sql = <<-EOQ
+  select
+      k.name as id,
+      k.title,
+      jsonb_build_object(
+        'Name', k.name,
+        'Location', k.location,
+        'Project', k.project,
+        'Self Link', k.self_link
+      ) as properties
+    from
+      gcp_pubsub_topic p,
+      gcp_pubsub_subscription k
+    where
+      p.name = k.topic_name 
+      and p.name = $1;
+  EOQ
+
+  param "name" {}
+}
+
+edge "gcp_pubsub_topic_to_pubsub_subscription_edge" {
+  title = "subscribed to"
+
+  sql = <<-EOQ
+  select
+      s.name as to_id,
+      t.name as from_id,
+      jsonb_build_object(
+        'Name', s.name,
+        'Location', s.location,
+        'Project', s.project,
+        'Self Link', s.self_link
+      ) as properties
+    from
+      gcp_pubsub_topic t,
+      gcp_pubsub_subscription s
+    where
+      t.name = s.topic_name 
+      and t.name = $1;
+  EOQ
+
+  param "name" {}
+}
+
+node "gcp_pubsub_topic_to_pubsub_snapshot_node" {
+  category = category.gcp_pubsub_subscription
+
+  sql = <<-EOQ
+  select
+      k.name as id,
+      k.title,
+      jsonb_build_object(
+        'Name', k.name,
+        'Location', k.location,
+        'Project', k.project,
+        'Self Link', k.self_link
+      ) as properties
+    from
+      gcp_pubsub_topic p,
+      gcp_pubsub_snapshot k,
+      gcp_pubsub_subscription s
+    where
+      p.name = s.topic_name 
+      and s.topic_name = k.topic_name
+      and p.name = $1;
+  EOQ
+
+  param "name" {}
+}
+
+edge "gcp_pubsub_topic_to_pubsub_snapshot_edge" {
+  title = "snapshot"
+
+  sql = <<-EOQ
+  select
+      k.name as to_id,
+      s.name as from_id,
+      jsonb_build_object(
+        'Name', k.name,
+        'Location', k.location,
+        'Project', k.project,
+        'Self Link', k.self_link
+      ) as properties
+    from
+      gcp_pubsub_topic p,
```

Comment:
@Priyanka-Chatterjee-2000, as discussed, this query will behave differently when we have multiple subscriptions and no snapshot; the join will return the duplicate data associated with the `snapshots` to the subscriptions having no `snapshots`
![image](https://user-images.githubusercontent.com/26873346/199507663-215a86b6-9980-46b2-a6eb-9b48953d77c3.png)

![image](https://user-images.githubusercontent.com/26873346/199507324-db92f1c3-3921-46ed-8278-aa465471e439.png)

**How do we address this?**
As per the API doc, we should link the snapshot to the topic
https://pkg.go.dev/google.golang.org/api/pubsub/v1#Snapshot.Topic